drizzle-orm/src/sql/expressions/conditions.ts

```diff
@@ -204,22 +393,116 @@ export function notInArray(
 	return sql`${column} not in ${bindIfParam(values, column)}`;
 }
 
+
+/**
+ * Test whether an expression is NULL. By the SQL standard,
+ * NULL is neither equal nor not equal to itself, so
+ * it's recommended to use `isNull` and `notIsNull` for
+ * comparisons to NULL.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Select cars that have no discontinuedAt date.
+ * db.select(cars)
+ *   .where(
+ *     isNull(cars.discontinuedAt))
+ * ```
+ *
+ * @see isNotNull for the inverse of this test
+ */
 export function isNull(column: AnyColumn | Placeholder | SQLWrapper): SQL {
 	return sql`${column} is null`;
 }
 
+/**
+ * Test whether an expression is not NULL. By the SQL standard,
+ * NULL is neither equal nor not equal to itself, so
+ * it's recommended to use `isNull` and `notIsNull` for
+ * comparisons to NULL.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Select cars that have been discontinued.
+ * db.select(cars)
+ *   .where(
+ *     isNotNull(cars.discontinuedAt))
+ * ```
+ *
+ * @see isNull for the inverse of this test
+ */
 export function isNotNull(column: AnyColumn | Placeholder | SQLWrapper): SQL {
 	return sql`${column} is not null`;
 }
 
+/**
+ * Test whether a subquery evaluates to have any rows.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Users whose `homeCity` column has a match in a cities
+ * // table.
+ * db
+ *   .select()
+ *   .from(users)
+ *   .where(
+ *     exists(db.select()
+ *      .from(cities)
+ *      .where(eq(users.homeCity, cities.id))));
+ * ```
+ *
+ * @see notExists for the inverse of this test
+ */
 export function exists(subquery: SQLWrapper): SQL {
 	return sql`exists (${subquery})`;
 }
 
+/**
+ * Test whether a subquery doesn't include any result
+ * rows.
+ *
+ * ## Examples
+ *
+ * ```ts
+ * // Users whose `homeCity` column doesn't match
+ * // a row in the cities table.
+ * db
+ *   .select()
+ *   .from(users)
+ *   .where(
+ *     notExists(db.select()
+ *      .from(cities)
+ *      .where(eq(users.homeCity, cities.id))));
```

Comment:
should `notExists` be here instead of `eq`?

Reply:
Same as the other one, there's a `notExists` and this is the subquery, I think we could totally make the subquery simpler, though.

Reply:
agree here also. Just didn't see it, sorry